src/key.rs

```diff
@@ -26,7 +27,20 @@ use Verification;
 use constants;
 use ffi::{self, CPtr};
 
-/// Secret 256-bit key used as `x` in an ECDSA signature
+/// Secret 256-bit key used as `x` in an ECDSA signature.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// # #[cfg(feature="rand")] {
+/// use secp256k1::{rand, Secp256k1, SecretKey};
+///
+/// let secp = Secp256k1::new();
+/// let secret_key = SecretKey::new(&mut rand::thread_rng());
+/// # }
```

Comment:
Nice idea with the scope. Untested idea that might also work (note the `#!`):

```suggestion
/// # #![cfg(feature="rand")]
/// use secp256k1::{rand, Secp256k1, SecretKey};
///
/// let secp = Secp256k1::new();
/// let secret_key = SecretKey::new(&mut rand::thread_rng());
```

Reply:
This doesn't work with Rust 1.29 unfortunately. I think, IIRC, it starts working at 1.34. I'll add a TODO to remove them after bump of MSRV.

Reply:
Fail! I did not 'note the `!#`' even when you told me too. It is _hard_ to get good help :)
Updating this PR to include this suggestion.

Reply:
Bother, and I thought the saying was 'if it builds ship it' ... turns out the `!#` suggestion builds just fine but doesn't run.

Reply:
Oh that is weird. Should go back to testing my suggestions before making them!

Reply:
Not at all, open suggestions lead to leaning stuff. And since they are cheaper to make than tested suggestions it leads to more information being shared, IMO.